### DISCUSSION TITLE
Teslamate on Qnap NAS

### DISCUSSION THREAD

**Discussion by maxgajo960** [teslamate-org/teslamate#2553] [General]
Hi, after doing all the steps and installed on the Qnap NAS, the Teslamate-1 app doesn't start and do not create the link to the webpage. Any idea how to fix it?
Thank you very much!
Max
![image](https://user-images.githubusercontent.com/97521720/159571625-fbbb70f6-e9ca-45ec-b5e6-cc8a44e910cf.png)

**Comment by Airosus** [comment]
Same issue here, running on a TS-230 QNAP NAS

**Reply by maxgajo960** [reply]
Any fix?

**Reply by Airosus** [reply]
Try adding a space after the “=“ and before your password in each of the three password lines. However, I’m still unable to view and data on grafana as I’m getting a “metric request error” on there (links now work at least)

**Reply by maxgajo960** [reply]
Ok, thanks I'll try. But tell me which encryption key and which password do I have to add in the script under the first "environment"? Moreover which password do I have to set for grafana and postgres? Tks, Max.

**Reply by maxgajo960** [reply]
I added the Tesla Tocken and I added the password in both grafana and postgres, the app starts but then stop, in loop.

**Reply by maxgajo960** [reply]
I cannot access even to :3000 (grafana)

**Reply by Airosus** [reply]
I can't get it to work at all - does anyone have a step-by-step guide for how to set this up on a QNAP/within Container Station?

**Comment by metawatchkalle** [comment]
I have exactly the same problem. No link available!  My workaround is to reboot the QNAP. In my case, after rebooting the NAS the link is available.

**Comment by maxgajo960** [comment]
Lucky you, i did it so many times but unsuccessfully...

**Reply by Eddy2222L** [reply]
Hello,

I encounter the same problem.  Have you solved it?

Thank you.

**Comment by metawatchkalle** [comment]
This is my docker-compose.yml and it worked for me for several month's on a QNAP TS-251 (32bitARM)
The only thing you have to do is to insert a self-chosen encryption-key.
I use the old grafana-version 1.26 because of the working map-dot. Unfortunatelly this version has no tyre-pressure on the grafana dashboards.

`version: "3"

services:
  teslamate:
    image: teslamate/teslamate:edge
    restart: always
    environment:
      - ENCRYPTION_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
      - DATABASE_USER=teslamate
      - DATABASE_PASS=secret
      - DATABASE_POOL_SIZE=8
      - DATABASE_TIMEOUT=90000
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:14-alpine
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=secret
      - POSTGRES_DB=teslamate
    ports:
      - 5432:5432
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:1.26
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=secret
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    ports:
      - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:
`

**Reply by Eddy2222L** [reply]
Thanks a lot for the reply and advice.  I replaced the #comment with a long key and now the Teslamate container is successfully installed (Database is running, both the 3000 and 4000 links are available) in my QNAP.

**Comment by Airosus** [comment]
Anyone have a step-by-step from A to Z for how to install this on a QNAP?

**Reply by metawatchkalle** [reply]
Qnap Model ?

**Reply by Eddy2222L** [reply]
Hi Airosus,

You can find the steps in this forum/thread https://www.reddit.com/r/TeslaMate/comments/z7scmc/reunning_teslamate_on_qnap_no_link_for_teslamate/

**Reply by Airosus** [reply]
TS-230 QNAP
